bugsnag-plugin-android-ndk/src/test/cpp/test_bsg_event.c

```diff
@@ -18,6 +31,106 @@ TEST test_event_context(void) {
     PASS();
 }
 
+TEST test_event_jailbroken(void) {
+    bugsnag_event *event = init_event();
+    ASSERT(bugsnag_app_get_jailbroken(event));
+    bugsnag_app_set_jailbroken(event, false);
+    ASSERT_FALSE(bugsnag_app_get_jailbroken(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_id(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_STR_EQ("my-id-123", event->device.id);
+    bugsnag_app_set_id(event, "SomeId");
+    ASSERT_STR_EQ("SomeId", bugsnag_app_get_id(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_locale(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_STR_EQ("en", event->device.locale);
+    bugsnag_app_set_locale(event, "hue");
+    ASSERT_STR_EQ("hue", bugsnag_app_get_locale(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_manufacturer(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_STR_EQ("Google", event->device.manufacturer);
+    bugsnag_app_set_manufacturer(event, "Apple");
+    ASSERT_STR_EQ("Apple", bugsnag_app_get_manufacturer(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_model(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_STR_EQ("Nexus", event->device.model);
+    bugsnag_app_set_model(event, "Pixel");
+    ASSERT_STR_EQ("Pixel", bugsnag_app_get_model(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_os_version(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_STR_EQ("9.1", event->device.os_version);
+    bugsnag_app_set_os_version(event, "7.0");
+    ASSERT_STR_EQ("7.0", bugsnag_app_get_os_version(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_total_memory(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_EQ(1095092340, event->device.total_memory);
+    bugsnag_app_set_total_memory(event, 200923409);
+    ASSERT_EQ(200923409, bugsnag_app_get_total_memory(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_orientation(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_STR_EQ("portrait", event->device.orientation);
+    bugsnag_app_set_orientation(event, "landscape");
+    ASSERT_STR_EQ("landscape", bugsnag_app_get_orientation(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_time(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_EQ(7609, event->device.time);
+    bugsnag_app_set_time(event, 1509);
+    ASSERT_EQ(1509, bugsnag_app_get_time(event));
+    free(event);
+    PASS();
+}
+
+TEST test_event_os_name(void) {
+    bugsnag_event *event = init_event();
+    ASSERT_STR_EQ("android", event->device.os_name);
+    bugsnag_app_set_os_name(event, "samsung");
+    ASSERT_STR_EQ("samsung", bugsnag_app_get_os_name(event));
+    free(event);
+    PASS();
+}
+
 SUITE(event_mutators) {
```

Comment:
Missing cpu_abi test to make this check all `device` fields.

Reply:
`cpu_abi` was left out of this PR as it will require a bit more thought since it's a struct, and the changeset was already quite large. I plan on addressing this in a separate PR.